### DISCUSSION TITLE
disconnect from useDisconnect is not closing metamask session.

### DISCUSSION THREAD

**Discussion by mtajammulzia** [wevm/wagmi#922] [Q&A] [answered]
Hello,

I have been integrating wagmi with our app and can't replicate a behaviour shown in wagmi examples and connectkit example.

When I disconnect my wallet with disconnect from useDisconnect hook and connect back I do not see any prompt from metamask. It just connects to the app instantly until I manually disconnect my app from "Connected Sites" tab from metamask. 
This is not the case with wagmi and conenctkit examples.

My config is as follows:

```js
const { chains, provider, webSocketProvider } = configureChains(
  [chain.rinkeby],
  [infuraProvider({ apiKey: "<key here>" })]
);

const client = createClient({
  autoConnect: true,
  connectors: [
    new MetaMaskConnector({
      chains,
      options: {
        shimDisconnect: true,
      },
    }),
    new WalletConnectConnector({
      chains,
      options: {
        qrcode: true,
      },
    }),
  ],
  provider,
  webSocketProvider,
});
```

Any ideas why this may be happening?

**Comment by tmm** [comment, marked as answer]
MetaMask [doesn't support programmatic disconnect](https://github.com/MetaMask/metamask-extension/issues/10353#issuecomment-1011129684). wagmi simulates this with the [`shimDisconnect`](https://wagmi.sh/docs/connectors/metaMask#shimdisconnect=) and [`UNSTABLE_shimOnConnectSelectAccount`](https://wagmi.sh/docs/connectors/metaMask#unstable_shimonconnectselectaccount=) flags.

Try adding `UNSTABLE_shimOnConnectSelectAccount: true` to `MetaMaskConnector.options` if you want to be prompted to choose an account after disconnecting.

**Comment by mtajammulzia** [comment]
Thanks @tmm I was trying with `shimDisconnect: true` which didn't work, using `UNSTABLE_shimOnConnectSelectAccount: true` works.

**Reply by tmm** [reply]
no problem! you probably want to use both `shimDisconnect` and `UNSTABLE_shimOnConnectSelectAccount` together for optimal results!

**Reply by mtajammulzia** [reply]
Alright will do, thanks!
